### DISCUSSION TITLE
Comets and Minor Planets

### DISCUSSION THREAD

**Discussion by Bernmeister** [XEphem/XEphem#55] [General]
FYI: The Minor Planet Center which provides data for computing minor planets and comets has stopped providing data for minor planets (in XEphem format). I have given a run-down of the scenario over on the `Skyfield` [discussion](https://github.com/skyfielders/python-skyfield/discussions/747) board for those interested.

**Comment by rmathar** [comment]
The Java code in https://doi.org/10.5281/zenodo.18965036 translates the MPC data files to the xephem format. Earlier versions of the code are in https://vixra.org/abs/1802.0035 (but vixra does not allow more than 5 updates so it is obsolete with some respects). Using this code allows you to be independent on the MPC actions to construct the xephem format.

**Comment by Bernmeister** [comment]
FYI: The original discussion has been marked as (somewhat) [answered](https://github.com/skyfielders/python-skyfield/discussions/747#discussioncomment-3470293).

**Comment by dave-kaye** [comment]
@brandon-rhodes to prove I am an equal opportunity pain-in-the-as*, I have sent three help requests to MPC to clear this issue.  The MPC web pages mentioned in the Skyfield post do exist and appear to be working as expected. This is the latest communication with the MPC Helpdesk.

--------------------------------------------------------------------------------------------------------

**MPCHLP-3509 RE: MPCHLP-3455 and MPCHLP-3377 orbital elements datasets in XEphem**
From:  "General Support"
To: *********
Date: May 6, 2024 4:51:29 PM
—-—-—-——-—-—-—
Michael Rudenko commented:

Thank you for brining this up again. The Bright list is only updated annually. The Critical list rarely changes.
There have been no new distant objects announced, so the Distant list has not been updated. The Unusual
list was working off of a stale data file, which has now been fixed so that should get updated daily if any new
unusual objects are observed.

Michael Rudenko resolved this as Done.

------- Original Issue Description -------
**RE: MPCHLP-3455 Resolved and MPCHLP-3377 Resolved orbital elements datasets in XEphem**

I continue to monitor four datasets that do not appear to be consistently updating. The main MPC data page
found at [www.minorplanetcenter.net/data](url) has a number of datasets available in the first table under the
Orbits heading. These datasets are further refined and converted to multiple, specialized software formats at
www.minorplanetcenter.net/iau/Ephemerides/SoftwareEls.html . In particular, I am interested in the datasets
converted to XEphem format found at www.minorplanetcenter.net/iau/Ephemerides/Soft03.html . It does
appear that updating one software version's dataset means that the dataset for all the software versions is
updated simultaneously.

The most recent communication from MPC Helpdesk was on 04/24/2024 which stated that all the datasets
had been set to update daily. These are the present results:

_Distant_
04/01/24 dataset epoch changed from 2/25/2023 to 3/31/2024 and modified data
04/01/24 -> 04/16/24 dataset remained the same
04/17/24 dataset modified data
04/17/24 -> 04/22/24 dataset remained the same
04/22/24 -> 04/25/24 dataset modified data somewhere within this period
04/25/24 -> 05/05/24 dataset remained the same

_Unusual_
04/01/24 -> 04/16/24 dataset remained the same
04/17/24 dataset modified data (1 line)
04/17/24 -> 04/22/24 dataset remained the same
04/22/24 -> 04/25/24 dataset epoch changed from 2/25/2023 to 3/31/2024 and modified data somewhere within this period
04/25/24 -> 05/05/24 dataset remained the same

_Critical List_
04/01/24 -> 04/16/24 dataset remained the same
04/17/24 dataset epoch changed from 5/23/2014 to 3/31/2024 and modified data
04/17/24 -> 04/22/24 dataset remained the same
04/22/24 -> 04/25/24 dataset modified data somewhere within this period
04/25/24 -> 05/05/24 dataset remained the same

_Bright_
04/01/24 -> 05/05/24 dataset remained the same with epoch 9/13/2023

**Results Summary:**

• The datasets Distant, Unusual and Critical List have been successfully updated to the present time period
and the epoch of 3/31/2024 that is found in the main MPCORB database. However, the communication of
4/24/24 indicated that they would update daily. This does not appear to be the case; they have remained the
same since 4/25/24.
• The dataset Bright has not been modified or updated during this period. There are several differing sources
of information that perhaps the Bright dataset is no longer operational, but it has existed for longer than its
supposed demise.
• The dataset Comets has been successfully updating daily during this time period.

Thank you for all your continuing efforts on these datasets.

---------------------------------------------------------------------------------------------------------

I disagree with the file update frequency of when new objects are sighted, since I see differences in location parameters from the main database on a daily basis.  But the differences are for very few objects and are very small - usually one/two units of the last significant digit of one/two orbital parameters. This appears to be as good as will be achievable at this time. So far, the datasets were updated directly on the days of replies to help desk requests. The key is to see if the files do actually update in the future, especially when the main database changes epochs.  

Overall, I think the MPC files are now operating and usable within XEphem.

**Comment by Bernmeister** [comment]
@dave-kaye et al: FYI, for minor planets I have switched to the [Asteroid Orbital Elements Database](https://asteroid.lowell.edu/astorb) and for comets the [Comet Observation Database](https://cobs.si/)[](https://cobs.si).

**Comment by dave-kaye** [comment]
I have revisited the MPC orbital elements datasets in XEphem, looking at the datasets on 07/15/24.  The results are as follows:

_Distant_
05/05/24 -> 07/15/24 dataset updated within this period, appears to have updated by at least 07/06/24

_Unusual_
05/05/24 -> 07/15/24 dataset updated within this period, appears to have updated by at least 07/12/24

_Critical List_
05/05/24 -> 07/15/24 dataset updated within this period

_Bright_
05/05/24 -> 07/15/24 dataset remained the same

_Comets_
05/05/24 -> 07/15/24 dataset updated within this period, appears to have updated by at least 07/14/24

It appears the XEphem datasets are successfully updating on the MPC website. The comets dataset is updating daily. The remainder are updating - though exact timing is not known, several appear to have been within a few days of the download. Only bright dataset is not updating, but that was communicated by MPC as updating only once per year.

**Comment by dave-kaye** [comment]
I have checked the Minor Planet Center data files used in Xephem to determine if they are continuing to be updated.  The check was performed on 9/5/25.  The results are as follows:

_Distant_
09/03/25 data set updated 05/05/25

_Unusual_
09/03/25 data set updated 05/05/25

_Critical List_
09/03/25 data set updated 05/05/25

_Bright_
09/03/25 data set updated 09/13/23 - i.e. not recently updated

_Comets_
09/03/25 data set updated 09/02/25

Overall, the data sets are being updated at a level of several times per year.  This appears to be sufficient since changes are minor. The exception is Bright which is supposed to be updated yearly but has not been updated since September 2023.

The Comets database is updating daily, as the MPC stated it would.

**Comment by dave-kaye** [comment]
MPC was notified of the issue with the Bright data set.  They have updated the link to the data set.  The results are now:

_Bright_
09/08/25 data set updated 05/05/25

**Comment by dave-kaye** [comment]
I have checked the Minor Planet Center data files used in XEphem to determine if they are being updated.  The results on 05/02/2026 were:

_Distant_
05/02/2026 data set updated 11/21/2025

_Unusual_
05/02/2026 data set updated 11/21/2025

_Critical List_
05/02/2026 data set updated 11/21/2025

_Bright_
05/02/2026 data set updated 11/21/2025

_Comets_
05/02/2026 data set updated 05/02/2026

The data continues to be updated at a level of several times a year.  The comets database is being updated daily.  All data sets have been updated since the last check.

**Reply by brandon-rhodes** [reply]
Thanks for the update! I'm glad they are continuing to update those data files.
